lib/ansible/executor/task_executor.py

```diff
@@ -399,6 +396,34 @@ def _run_loop(self, items):
 
         return results
 
+    def _calculate_delegate_to(self, templar, variables):
+        if self._task.delegate_to:
+            delegated_host_name = templar.template(self._task.delegate_to, fail_on_undefined=False)
+            delegated_host = self._variable_manager._inventory.get_host(delegated_host_name)
+            if delegated_host is None:
+                for h in self._variable_manager._inventory.get_hosts(ignore_limits=True, ignore_restrictions=True):
```

Comment:
having to use _inventory from VM seems wrong .. but we suck at keeping private .. private, but tempted to move this to 'delegated_host, delegatedvars = vm.get_delegated_host_and_Vars(task.delegate_to)'  call

Reply:
Yeah, we're already passing VM into the worker, but we don't independently pass the inventory.  To avoid further overloading the `__init__` method for `WorkerProcess` and `TaskExecutor`, I chose to just use the dangled copy.  Since it's our code, if we ever change something later it's easy enough to address it later.

I was also concerned about any potential pickling overhead of passing it 2 times effectively.

Reply:
I'm not against moving it back to VM as a method we can call.